sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -813,7 +813,7 @@ uint8_t gcoap_op_state(void)
 int gcoap_get_resource_list(void *buf, size_t maxlen, uint8_t cf)
 {
     assert(cf == COAP_CT_LINK_FORMAT);
-#ifndef DEVELHELP
```

Comment:
IMHO both are superfluous, the `cf` parameter isn't used other than in the `assert`, I would remove the preprocessor regardless of `DEVELHELP` or `NDEBUG` and leave the `(void)cf;` standalone.

Make the code more clean and result will also be the same (code size wise and so on).

TL;DR: better remove `#if[n]def`.

Reply:
Well…I actually disagree since the `#ifdef` makes it more clear that the variable is **only** used in the assert-statement but since you seem to see this differently I changed it anyhow.